### ISSUE TITLE
F16v3 crashes on upload

### ISSUE THREAD

**Issue by Cboehm00** [smeighan/xLights#896]
When I upload output data to F16v3 the whole controller freezes and must be factory reset in order to access string ports. I've narrowed it down to this only happening when the controller is set up for 2 expansion boards. If its set for mainboard only, or 1 expansion the upload will work fine.

**Comment by AzGilrock** [comment]
Well I know zero about that upload but since you can always do it manually its low priority.  Plus there should be no way that an outside interface can lock up the controller so that would need to be fixed by Dave.

**Comment by keithsw1111** [comment]
Personally I think there are two problems here. Xlights is obviously sending to the falcon something problematic but obviously the falcon shouldn't crash when it receives it. We are both aware of the problem and looking into it.

**Comment by keithsw1111** [comment]
I believe i have this fixed for .23

**Comment by Cboehm00** [comment]
Hello,
I downloaded .23 today and tried it out to see if my problem with uploading the output to the falcon has been resolved.
Seems it works fine now if I have it set up for main board only or with 1 expansion.  If I set it up for 2 expansions and upload the output to the falcon I get the same solid blue oled screen and a frozen controller when I try to access the string ports.

Here's what I did:
I added 1 element only to my layout, I set it up to be on port 44 which would be a second expansion.  Then I uploaded and the upload seems to go through and everything seems ok until I try to access the string ports in the controller.  That's when it freezes and turns the oled solid blue.  I then reset the controller to factory settings and set that element to port 17, uploaded, and everything works great.
I thought maybe it was specific to my computer, not so.  I tried 3 different computers, I tried the 32 bit and the 64 bit and get the same result.


Thanks for all your hard work!
Corry




On Aug 13, 2017, at 3:50 PM, Keith Westley <notifications@github.com<mailto:notifications@github.com>> wrote:


I believe i have this fixed for .23

**Comment by keithsw1111** [comment]
Can you package your logs (tools menu) and then send them to us.

**Comment by Cboehm00** [comment]
attached is my log files.

________________________________

**Comment by keithsw1111** [comment]
Replying with your log files attached to an email doesnt attach them to the case ... you need to do that on the github web page. There are additional changes coming in .24 which may also help.

**Comment by Cboehm00** [comment]
Ok, I'll wait on .24 then. Thanks.

Sent from my iPhone

On Aug 20, 2017, at 2:37 PM, Keith Westley <notifications@github.com<mailto:notifications@github.com>> wrote:


Replying with your log files attached to an email doesnt attach them to the case ... you need to do that on the github web page. There are additional changes coming in .24 which may also help.

**Comment by keithsw1111** [comment]
I have spent some time with Dave over the past few days and we now have it updating without crashing. This will be in the .26 release.
